### ISSUE TITLE
TypeScript typings

### ISSUE THREAD

**Issue by nucleartide** [poteto/ember-changeset#209]

**Comment by nucleartide** [comment]
See https://github.com/poteto/validated-proxy.

**Comment by nucleartide** [comment]
Now that [ember-cli-typescript](https://github.com/typed-ember/ember-cli-typescript#using-ember-cli-typescript-with-ember-cli-addons) has some documentation about usage in addons, we should consider moving our Flow comments over to TypeScript.

**Comment by ryanlabouve** [comment]
Yo @nucleartide, I'd love to help move `ember-changeset` towards typescript instead of flow. Are you still wanting help with this?

**Comment by nucleartide** [comment]
@ryanlabouve Yeah I'd love some help! In particular, I feel like the TypeScript ember-data typings would suss out a lot of the bugs between ED relationships and nested keys.

If you have the time, lemme know how you plan to approach things. I'm very open to a PR for this.

**Comment by ryanlabouve** [comment]
Perhaps we can remove flow in 1 PR and then open up an epic issue to allow us to introduce typescript typings in small chunks?

**Comment by nucleartide** [comment]
We should be able to introduce TypeScript without touching Flow, since all the Flow annotations are wrapped in comments. So both Flow and TypeScript would live side-by-side until we remove all the Flow annotations.
